src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -318,18 +327,57 @@ private void HandleObservableInstrumentCallbackError(TraceEvent obj)
         private void HandleMultipleSessionsNotSupportedError(TraceEvent obj)
         {
             string runningSessionId = (string)obj.PayloadValue(0);
-            if (runningSessionId == _metricsEventSourceSessionId)
+            if (runningSessionId == _sessionId)
             {
                 // If our session is the one that is running then the error is not for us,
                 // it is for some other session that came later
                 return;
             }
             _renderer.SetErrorText(
-                "Error: Another metrics collection session is already in progress for the target process, perhaps from another tool?" + Environment.NewLine +
+                "Error: Another metrics collection session is already in progress for the target process." + Environment.NewLine +
                 "Concurrent sessions are not supported.");
             _shouldExit.TrySetResult((int)ReturnCode.SessionCreationError);
         }
 
+        private void HandleMultipleSessionsConfiguredIncorrectlyError(TraceEvent obj)
```

Comment:
This roughly looks like the same in the TraceEventExtensions class. Consider changing that one to be `bool TryCreateSharedSessionConfiguredIncorrectlyMessage(TraceEvent obj, string clientId, out string message)` which will return true if the `clientId` matches that of the `TraceEvent` object; link that source file into both M.D.M.EventPipe and dotnet-counters project. Each handler can then set their version of `inactiveSharedSessions` and related fields as they need. This would minimize the code duplication.

Reply:
I couldn't tell if you wanted this shared method to stay in TraceEventExtensions or not so I've left it there for now - let me know if you'd like it relocated to a more centralized location